System/Posix/Directory/Common.hsc

```diff
@@ -36,12 +39,23 @@ module System.Posix.Directory.Common (
 import System.Posix.Types
 import Foreign
 import Foreign.C
+import Data.Functor ((<$>))
 
 newtype DirStream = DirStream (Ptr CDir)
 
 data {-# CTYPE "DIR" #-} CDir
 data {-# CTYPE "struct dirent" #-} CDirent
 
+#ifdef HAVE_FDOPENDIR
+-- | @openDirStreamFd fd@ calls @fdopendir@ to obtain a
```

Comment:
This should document that you must not close the file descriptor afterwards with a [link to the POSIX documentation](https://pubs.opengroup.org/onlinepubs/9699919799/functions/fdopendir.html).

Reply:
Done